### PR TITLE
Trying to change the group name to an already existing name now fails

### PR DESCRIPTION
**Pull request by MoritzHahn1337**

### CODE REVIEW

**Review by lkuchenb — COMMENTED** [review]
We should be enforcing this in the database model

**Comment by lkuchenb** [comment]
@MoritzHahn1337 I've pushed the DB change against this branch. Could you change the handling in the backend function such that it now catches the database error?

**Comment by MoritzHahn1337** [comment]
Will do, but probably won't get to it until monday

**Comment by MoritzHahn1337** [comment]
When trying to change the database via a db.commit(),  a `sqlalchemy.exc.IntegrityError: (psycopg2.errors.UniqueViolation)` is thrown. As db.commit() is executed on a return statement, I expected `try: ... except: ...` to catch the error, but it does not:

```
try:
    target_group.name = new_group_name
    return HTTPNoContent()
except:
    response_body = []
    err["message"] = "This group name does already exist."
    response_body.append(err)
    raise HTTPBadRequest(json=response_body)```
    
Still throws the same error.

**Comment by MoritzHahn1337** [comment]
After trying some more, I did not find an easy way to catch a database error and reverted to catching a naming collision manually for the time beeing.

**Comment by MoritzHahn1337** [comment]
db.flush() did do the trick.

File: tests/integration/test_groupnameupdate.py [hunk 1]
```diff
@@ -8,51 +8,29 @@
...
-        ("own_as_regular_user"       , "user_a"          , "group_x_id"   , 403),
-        ("foreign_as_regular_user"   , "user_a"          , "group_y_id"   , 403)
+        # TEST_NAME                    EXEC_USER           TARGET_GRP       NEW_GROUPNAME    RESP
+        ("foreign_as_admin"            , "admin"           , "group_x_id"   , "fancy_group"  , 204),
+        ("own_as_group_admin"          , "group_x_admin"   , "group_x_id"   , "fancy_group"  , 403),
+        ("foreign_as_group_admin"      , "group_x_admin"   , "group_y_id"   , "fancy_group"  , 403),
+        ("invalid_group"               , "admin"           , "duckburgh"    , "fancy_group"  , 403),
+        ("own_as_regular_user"         , "user_a"          , "group_x_id"   , "fancy_group"  , 403),
+        ("foreign_as_regular_user"     , "user_a"          , "group_y_id"   , "fancy_group"  , 403),
+        # this throws an IntegrityError instead of 400
```
Comment (cschu) [inline tests/integration/test_groupnameupdate.py:18]:
```suggestion

```

Comment (cschu) [reply tests/integration/test_groupnameupdate.py:18]: no longer the case :)

File: datameta/api/groups.py [hunk 2]
```diff
@@ -129,9 +130,20 @@ def put(request: Request):
...
+            target_group.name = new_group_name
+            db.flush()
+        except IntegrityError:
+            response_body = []
+            err = {
+                "exception": "ValidationError",
+                "message" : "This group name does already exist."
+            } 
+            response_body.append(err)
+            raise HTTPBadRequest(json=response_body)
```
Comment (lkuchenb) [inline datameta/api/groups.py:146]:
```suggestion
            raise errors.get_validation_error("A group with that name already exists.")
```

**Review by lkuchenb — CHANGES_REQUESTED** [review]
Thanks, looks good to me. One minor fix request below.

**Review by lkuchenb — APPROVED** [review]
Good to go!

**Pull request merged**